From da6d1f07a007a70c5827d6855c3fbe4ab3d205bc Mon Sep 17 00:00:00 2001
From: Kentaro Hayashi <hayashi@clear-code.com>
Date: Thu, 1 May 2025 18:25:18 +0900
Subject: [PATCH] Do not install iconv (GPLv3)

Signed-off-by: Kentaro Hayashi <hayashi@clear-code.com>
---
 Makefile.in | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 5ddabe1..5f57dfb 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -32,7 +32,6 @@ SHELL = @SHELL@
 all : lib/localcharset.h force
 	cd lib && $(MAKE) all
 	cd srclib && $(MAKE) all
-	cd src && $(MAKE) all
 	cd po && $(MAKE) all
 	cd man && $(MAKE) all
 	if test -d tests; then cd tests && $(MAKE) all; fi
@@ -51,7 +50,6 @@ install : lib/localcharset.h force
 	cd libcharset && $(MAKE) install prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	cd lib && $(MAKE) install prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	cd srclib && $(MAKE) install prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
-	cd src && $(MAKE) install prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	if [ ! -d $(DESTDIR)$(includedir) ] ; then $(mkinstalldirs) $(DESTDIR)$(includedir) ; fi
 	$(INSTALL_DATA) include/iconv.h.inst $(DESTDIR)$(includedir)/iconv.h
 	cd po && $(MAKE) install prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)'
@@ -61,7 +59,6 @@ install-strip : lib/localcharset.h force
 	cd libcharset && $(MAKE) install-strip prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	cd lib && $(MAKE) install-strip prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	cd srclib && $(MAKE) install-strip prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
-	cd src && $(MAKE) install-strip prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	if [ ! -d $(DESTDIR)$(includedir) ] ; then $(mkinstalldirs) $(DESTDIR)$(includedir) ; fi
 	$(INSTALL_DATA) include/iconv.h.inst $(DESTDIR)$(includedir)/iconv.h
 	cd po && $(MAKE) install-strip prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)'
@@ -71,7 +68,6 @@ installdirs : force
 	cd libcharset && $(MAKE) installdirs prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	cd lib && $(MAKE) installdirs prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	cd srclib && $(MAKE) installdirs prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
-	cd src && $(MAKE) installdirs prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	if [ ! -d $(DESTDIR)$(includedir) ] ; then $(mkinstalldirs) $(DESTDIR)$(includedir) ; fi
 	cd po && $(MAKE) installdirs prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)'
 	cd man && $(MAKE) installdirs prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)' mandir='$(mandir)'
@@ -80,7 +76,6 @@ uninstall : force
 	cd libcharset && $(MAKE) uninstall prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	cd lib && $(MAKE) uninstall prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	cd srclib && $(MAKE) uninstall prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
-	cd src && $(MAKE) uninstall prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	$(RM) $(DESTDIR)$(includedir)/iconv.h
 	cd po && $(MAKE) uninstall prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)'
 	cd man && $(MAKE) uninstall prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)' mandir='$(mandir)'
@@ -89,7 +84,6 @@ check : all force
 	cd libcharset && $(MAKE) check
 	cd lib && $(MAKE) check
 	cd srclib && $(MAKE) check
-	cd src && $(MAKE) check
 	cd po && $(MAKE) check
 	cd man && $(MAKE) check
 	if test -d tests; then cd tests && $(MAKE) check; fi
@@ -99,7 +93,6 @@ mostlyclean : force
 	$(RM) lib/libcharset.* lib/localcharset.h
 	cd lib && $(MAKE) mostlyclean
 	cd srclib && $(MAKE) mostlyclean
-	cd src && $(MAKE) mostlyclean
 	cd po && $(MAKE) mostlyclean
 	cd man && $(MAKE) mostlyclean
 	if test -d tests; then cd tests && $(MAKE) mostlyclean; fi
@@ -110,7 +103,6 @@ clean : force
 	$(RM) lib/libcharset.* lib/localcharset.h
 	cd lib && $(MAKE) clean
 	cd srclib && $(MAKE) clean
-	cd src && $(MAKE) clean
 	cd po && $(MAKE) clean
 	cd man && $(MAKE) clean
 	if test -d tests; then cd tests && $(MAKE) clean; fi
@@ -121,7 +113,6 @@ distclean : force
 	$(RM) lib/libcharset.* lib/localcharset.h
 	cd lib && if test -f Makefile; then $(MAKE) distclean; fi
 	cd srclib && if test -f Makefile; then $(MAKE) distclean; fi
-	cd src && if test -f Makefile; then $(MAKE) distclean; fi
 	cd po && if test -f Makefile; then $(MAKE) distclean; fi
 	cd man && if test -f Makefile; then $(MAKE) distclean; fi
 	if test -d tests; then cd tests && if test -f Makefile; then $(MAKE) distclean; fi; fi
@@ -135,7 +126,6 @@ maintainer-clean : force
 	$(RM) lib/libcharset.* lib/localcharset.h
 	cd lib && if test -f Makefile; then $(MAKE) maintainer-clean; fi
 	cd srclib && if test -f Makefile; then $(MAKE) maintainer-clean; fi
-	cd src && if test -f Makefile; then $(MAKE) maintainer-clean; fi
 	cd po && if test -f Makefile; then $(MAKE) maintainer-clean; fi
 	cd man && if test -f Makefile; then $(MAKE) maintainer-clean; fi
 	if test -d tests; then cd tests && if test -f Makefile; then $(MAKE) maintainer-clean; fi; fi
@@ -226,7 +216,6 @@ distdir : $(DISTFILES)
 	test -d '$(distdir)'/libcharset || mkdir '$(distdir)'/libcharset; cd libcharset && $(MAKE) distdir distdir='$(distdir)'/libcharset
 	test -d '$(distdir)'/lib || mkdir '$(distdir)'/lib; cd lib && $(MAKE) distdir distdir='$(distdir)'/lib
 	test -d '$(distdir)'/srclib || mkdir '$(distdir)'/srclib; cd srclib && $(MAKE) distdir distdir='$(distdir)'/srclib
-	test -d '$(distdir)'/src || mkdir '$(distdir)'/src; cd src && $(MAKE) distdir distdir='$(distdir)'/src
 	test -d '$(distdir)'/tests || mkdir '$(distdir)'/tests; cd tests && $(MAKE) distdir distdir='$(distdir)'/tests
 	test -d '$(distdir)'/po || mkdir '$(distdir)'/po; cd po && $(MAKE) distdir distdir='$(distdir)'/po
 	test -d '$(distdir)'/man || mkdir '$(distdir)'/man; cd man && $(MAKE) distdir distdir='$(distdir)'/man
-- 
2.49.0

