From 4fe0b8c4ad1b17b79a024858e2d054b12f8093a8 Mon Sep 17 00:00:00 2001
From: Kentaro Hayashi <hayashi@clear-code.com>
Date: Mon, 10 Mar 2025 20:15:42 +0900
Subject: [PATCH 2/2] Salvage missing l64a implementation for Android

Signed-off-by: Kentaro Hayashi <hayashi@clear-code.com>
---
 lib/salt.c | 35 +++++++++++++++++++++++++++++++++++
 1 file changed, 35 insertions(+)

diff --git a/lib/salt.c b/lib/salt.c
index 753e9d45..eea9c588 100644
--- a/lib/salt.c
+++ b/lib/salt.c
@@ -320,6 +320,41 @@ static /*@observer@*/void YESCRYPT_salt_cost_to_buf (char *buf, unsigned long co
 #endif /* USE_YESCRYPT */
 
 #if !USE_XCRYPT_GENSALT
+#if !defined(HAVE_L64A)
+static /*@observer@*/char *l64a (long value)
+{
+       static char buf[8];
+       char *s = buf;
+       int digit;
+       int i;
+
+       if (value < 0) {
+               errno = EINVAL;
+               return(NULL);
+       }
+
+       for (i = 0; value != 0 && i < 6; i++) {
+               digit = value & 0x3f;
+
+               if (digit < 2) {
+                       *s = digit + '.';
+               } else if (digit < 12) {
+                       *s = digit + '0' - 2;
+               } else if (digit < 38) {
+                       *s = digit + 'A' - 12;
+               } else {
+                       *s = digit + 'a' - 38;
+               }
+
+               value >>= 6;
+               s++;
+       }
+
+       *s = '\0';
+
+       return buf;
+}
+#endif /* !USE_XCRYPT_GENSALT && !defined(HAVE_L64A) */
 static /*@observer@*/const char *gensalt (size_t salt_size)
 {
 	static char salt[MAX_SALT_SIZE + 6];
-- 
2.47.2

