From 4c32c7952ecb0dd18b8c13e2c3fd628239691f7d Mon Sep 17 00:00:00 2001
From: Takuro Ashie <ashie@clear-code.com>
Date: Tue, 11 Mar 2025 17:07:26 +0900
Subject: [PATCH 6/7] Add missing vring_need_event()

---
 virtio.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/virtio.c b/virtio.c
index e6282b9..4a82b46 100644
--- a/virtio.c
+++ b/virtio.c
@@ -297,6 +297,11 @@ bool vu_queue_empty(struct vu_virtq *vq)
 	return vring_avail_idx(vq) == vq->last_avail_idx;
 }
 
+static inline int vring_need_event(uint16_t event_idx, uint16_t new_idx, uint16_t old)
+{
+	return (uint16_t)(new_idx - event_idx - 1) < (uint16_t)(new_idx - old);
+}
+
 /**
  * vring_can_notify() - Check if a notification can be sent
  * @dev:	Vhost-user device
-- 
2.34.1

